datafusion/optimizer/src/optimizer.rs

```diff
@@ -184,39 +185,12 @@ pub struct Optimizer {
     pub rules: Vec<Arc<dyn OptimizerRule + Send + Sync>>,
 }
 
-/// If a rule is with `ApplyOrder`, it means the optimizer will derive to handle children instead of
-/// recursively handling in rule.
-/// We just need handle a subtree pattern itself.
+/// Specifies how recursion for an `OptimizerRule` should be handled.
 ///
-/// Notice: **sometime** result after optimize still can be optimized, we need apply again.
```

Comment:
I do not think this comment is applicable anymore -- the optimizer handles the recursion internally as well as applying multiple optimizer passes